plugins/core/ui/src/components/builtin/ShellComponent.vue

```diff
@@ -40,19 +40,20 @@ export default {
     this.socket.send(JSON.stringify({ dim: { cols: term.cols, rows: term.rows } }));
 
     this.socket.on("message", (data) => {
-      term.write(new Uint8Array(Buffer.from(JSON.parse(data).data)));
+      term.write(new Uint8Array(Buffer.from(data)));
     });
 
     term.onData((data) => {
-      this.socket.send(JSON.stringify({ data }));
+      this.socket.send(JSON.stringify({ d: data }));
```

Comment:
Should send this as a buffer as well, all term data can go across the wire as binary. No reason to send json for text.